.github/workflows/run_test.yaml

```diff
@@ -0,0 +1,39 @@
+name: Pytest
+
+on: [push, pull_request]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.11"]
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Cache pip packages
+        uses: actions/cache@v3
+        with:
+          path: ~/.cache/pip  # Cache pip packages
+          key: ${{ runner.os }}-pip-${{ matrix.python-version }}-${{ hashFiles('**/requirements.txt') }}
+          restore-keys: |
+            ${{ runner.os }}-pip-${{ matrix.python-version }}-
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
```

Comment:
AFAIK it's just common standard to do it this way, so it fails closer to the actual source. I don't feel super strong about this

Reply:
Can do. In the end, the requirements will always be there, and it doesn't matter. But if they weren't there, I suspect getting an error later at the pytest stage because some dependencies are not installed and not because the requirements file is non-existent.